### ISSUE TITLE
HATEOAS AutoConfiguration does not work as expected

### ISSUE THREAD

**Issue by mvitz** [spring-projects/spring-boot#8135]
According to https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-developing-web-applications.html#boot-features-spring-hateoas it should not be required to add `@EnableHypermediaSupport` to on any configuration class.

If I follow this advice unfortunately the application does not produce any valid HAL response. If I add the annotation everything works as expected. Either this is a bug or the documentation should be improved.

I pusht an example application to show this behaviour at https://github.com/mvitz/spring-boot-hateoas-autoconfiguration-bug

**Comment by wilkinsona** [comment]
Your endpoint is returning a `List<UserResource>` which will be mapped to a JSON array. HAL doesn't support array responses – the response must be a JSON object. Arguably things are behaving more correctly with the auto-configured HATEOAS. I think, although I haven't double-checked, that this is due to the fact that your `@EnableHypermediaSupport` is switching off `HalObjectMapperConfigurer`

The `/users/{name}` endpoint which returns an Object, and therefore can be serialised to legal HAL, works fine with the auto-configured HATEOAS support:

```
$ curl -i localhost:8080/users/foo -H Accept:application/hal+json
HTTP/1.1 200
Content-Type: application/hal+json;charset=UTF-8
Transfer-Encoding: chunked
Date: Mon, 30 Jan 2017 13:48:38 GMT

{"_links":{"self":{"href":"http://localhost:8080/users/foo"}}}
```

In summary, you need to update your endpoint for returning all users so that it returns a `ResourceSupport` subclass rather than a `List`.
